### PR TITLE
#160994766 Implement Authorisation on the Frontend 

### PR DESCRIPTION
**Pull request by CEOehis**
#### What does this PR do?
Enable restricted access to privileged pages on the front-end

#### Description of Task to be completed?
- Create `ProtectedRoute` component
- Redirect unauthorized user to `login` page
- Implement `sign out` action
- Include relevant tests

#### What are the relevant pivotal tracker stories?
[#160994766](https://www.pivotaltracker.com/story/show/160994766)

### CODE REVIEW

File: src/reducers/auth.reducer.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import isEmpty from 'lodash/isEmpty';
+import { SET_CURRENT_USER, LOG_OUT_USER } from "../actions/action.types";
+
+const initialState = {
+  isAuthenticated: true,
```
Comment (emekafredy) [inline src/reducers/auth.reducer.js]: Hello, @CEOehis Nice job so far. Please, I'd like to know why isAuthenticated is set to true at the initial State. I thought it should be false

File: src/tests/reducers/auth.reducer.test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import auth from '../../reducers/auth.reducer';
+import * as types from '../../actions/action.types';
+
+describe('user reducer', () => {
+  it('should return initial state', () => {
+    expect(auth(undefined, {})).toEqual({
+      isAuthenticated: true,
```
Comment (emekafredy) [inline src/tests/reducers/auth.reducer.test.js]: Initial state here for isAuthenticated should be false too

**Review by emekafredy — CHANGES_REQUESTED** [review]
@CEOehis Please check the comments for requested changes. Nice one

File: src/reducers/auth.reducer.js (hunk 1 above)
Comment (CEOehis) [reply src/reducers/auth.reducer.js]: Thanks for pointing this out. I'll fix this ASAP.

File: src/tests/App.test.js [hunk 3]
```diff
@@ -1,16 +1,11 @@
...
+import toJson from 'enzyme-to-json';
 import App from "../App";
 
-it("renders without crashing", () => {
-  const div = document.createElement("div");
-  ReactDOM.render(<App />, div);
-  ReactDOM.unmountComponentAtNode(div);
-});
-
 it("create a snapshot", () => {
```
Comment (emmaadesile) [inline src/tests/App.test.js]: Line 12 should be `it(creates a snapshot)`

**Review by emekafredy — APPROVED** [review]
Looks Good

File: src/tests/App.test.js (hunk 3 above)
Comment (emmaadesile) [inline src/tests/App.test.js]: Line 12 should be `it(creates a snapshot)`

**Review by emmaadesile — CHANGES_REQUESTED** [review]

File: src/tests/App.test.js (hunk 3 above)
Comment (CEOehis) [reply src/tests/App.test.js]: I don't think it should be 'creates a snapshot'. The test is to 'create a snapshot'. It is more or less a test to check the interface.

File: src/actions/auth.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable import/prefer-default-export */
```
Comment (johnkingzy) [inline src/actions/auth.js:1]: I think this should be disabled generally

**Review by johnkingzy — APPROVED** [review]
Good job 👍

**Review by emmaadesile — APPROVED** [review]

**Pull request merged**